akka-distributed-data/src/main/scala/akka/cluster/ddata/ORMap.scala

```diff
@@ -100,8 +151,17 @@ final class ORMap[A, B <: ReplicatedData] private[akka] (
         "`ORMap.put` must not be used to replace an existing `ORSet` " +
           "value, because important history can be lost when replacing the `ORSet` and " +
           "undesired effects of merging will occur. Use `ORMultiMap` or `ORMap.updated` instead.")
-    else
```

Comment:
I don't think this is particular to the *causal CRTDs*. All are affected, because `ORSet` indexing `ORMap` is causal, so replacements of full values are always causing anomalies. However, the effect of this is probably most visible when the value in question is itself an `ORSet`. On the other hand working with `updated` is a bit harder, as you need to supply the transformation function, which is helped by pre-defined maps - the _friends_ of `ORMap`: `ORMultiMap`, `PNCounterMap` and `LWWMap`. So I would probably leave that as is or, for consistency, either extend it to all values or to all friends. WDYT?